File: 程序设计思维与实践作业/作业/Month1/编程题6-小H爱数学.py
def solve():
    cnt = 0
    tem = 1
    l, r, k = map(int, input().split())
    
    while True:
        if tem >= l and tem <= r:
            cnt += 1
            print(tem, end=' ')
        tem *= k
        if tem > r:
            break
    
    if cnt == 0:
        print(-1)

solve()

File: 程序设计思维与实践作业/作业/Month1/编程题6-小H爱数学.cpp
#include <iostream>
#include <algorithm>
#include <limits>
using namespace std;


void solve()
{
    size_t cnt = 0;
    size_t l, r, k;
    cin >> l >> r >> k;
    size_t tem = 1;
    while (true)
    {
        if (tem >= l && tem <= r)
        {
            cnt++;
            cout << tem << ' ';
        }
        if (numeric_limits<size_t>::max() / k < tem) break;
        tem *= k;
        if (tem > r) break;
    }
    if (cnt == 0) cout << -1;
}

int main()
{
    solve();
    return 0;
}